ocs_ci/utility/csr.py

```diff
@@ -187,10 +188,15 @@ def wait_for_all_nodes_csr_and_approve(timeout=900, sleep=10, expected_node_num=
         pending_csrs = get_pending_csr()
         if pending_csrs:
             approve_csrs(pending_csrs)
-        # In vSphere deployment it sometime happes that VM doesn't get ip and
+        # In vSphere deployment it sometime happens that VM doesn't get ip and
         # then we need to restart it to make our CI more stable and let the VM
-        # to get IP and continue with loading ignition config. The rester of
+        # to get IP and continue with loading ignition config. The restart of
         # the VMs happens only once in reboot_timeout (120 seconds).
         if vsphere_object and time.time() - start_time >= reboot_timeout:
             start_time = time.time()
-            vsphere_object.find_vms_without_ip_and_restart()
+            if not is_vms_without_ip:
+                vms_without_ip = vsphere_object.find_vms_without_ip()
```

Comment:
with your change: find_vms_without_ip is taking name, datacenter_name, cluster_name as arguments but you don't pass any of those.

Or am I missing something?

Reply:
missed that. added params in latest commit